module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -404,48 +425,83 @@ public function getConfig($function, $params = null)
-            $locationName,
+            $locationId,
             (array)($this->config['Holds']['excludeHoldLocations'] ?? [])
         );
     }
 
     /**
-     * Gets the location name from the /locations endpoint and sets
-     * the display name to discoveryDisplayName,  name, or code
-     * based on whichever is available first in that order.
+     * Gets locations from the /locations endpoint and sets
+     * an array of location IDs to display names.
+     * Display names are set from discoveryDisplayName, or name
+     * if discoveryDisplayName is not available.
      *
-     * @param string $locationId ID of a location from the
-     * /holdings-storage/holdings endpoint
+     * @return array
+     */
+    protected function getLocations()
+    {
+        $cacheKey = 'locationMap';
+        $locationMap = $this->getCachedData($cacheKey);
+        //die(var_dump($locationMap));
+        if (null === $locationMap) {
+            $locationMap = [];
+            foreach ($this->getPagedResults(
+                'locations', '/locations'
+            ) as $location) {
+                if (!empty($location->discoveryDisplayName)) {
+                    // discoveryDisplayName is optional
+                    $locationMap[$location->id] = $location->discoveryDisplayName;
+                } else {
+                    // name is required
+                    $locationMap[$location->id] = $location->name;
+                }
+            }
+        }
+        $this->putCachedData($cacheKey, $locationMap);
+        return $locationMap;
+    }
+
+
+    /**
+     * Get Inventory Location Name
      *
-     * @return string
+     * @param string $locationId UUID of item location
+     *
+     * @return string $locationName display name of location
      */
     protected function getLocationName($locationId)
     {
+        $locationMap = $this->getLocations();
         $locationName = '';
-        if (!empty($locationId)) {
+        if (array_key_exists($locationId, $locationMap)) {
+            $locationName = $locationMap[$locationId];
+        } else {
+            // if key is not found in cache, the location could have
+            // been added before the cache expired so check again
             $locationResponse = $this->makeRequest(
-                'GET',
-                '/locations/' . $locationId
+                'GET', '/locations/' . $locationId
             );
-            $location = json_decode($locationResponse->getBody());
-            if (!empty($location->discoveryDisplayName)) {
-                $locationName = $location->discoveryDisplayName;
-            } elseif (!empty($location->name)) {
-                $locationName = $location->name;
-            } elseif (!empty($location->code)) {
-                $locationName = $location->code;
+            if ($locationResponse->isSuccess()) {
+                $json = json_decode($locationResponse->getBody());
+                if (!empty($json->discoveryDisplayName)) {
```

Comment:
If you accept my proposed if..else simplification in the previous comment, you should probably apply the same logic here for consistency.